crates/client/src/lib.rs

```diff
@@ -485,18 +492,64 @@ impl<R: RegistryStorage, C: ContentStorage, N: NamespaceMapStorage> Client<R, C,
         tracing::info!("updating specific packages to latest checkpoint");
 
         let packages = packages.into_iter();
-        let mut updating = Vec::with_capacity(packages.len());
+        let mut namespaced: IndexMap<&str, Vec<&PackageName>> = IndexMap::new();
         for package in packages {
-            updating.push(
-                self.registry
-                    .load_package(self.api.get_warg_registry(), package)
-                    .await?
-                    .unwrap_or_else(|| PackageInfo::new(package.clone())),
-            );
+            let namespace = package.namespace();
+            let namespace_packages = namespaced.get_mut(namespace);
+            if let Some(nm_pkgs) = namespace_packages {
+                nm_pkgs.push(package);
+            } else {
+                namespaced.insert(namespace, vec![package]);
+            }
```

Comment:
nit: There is a nice pattern for this:

```suggestion
            namespaced.entry(namespace).or_default().push(package);
```